lib/src/presentation/widgets/app_drawer.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+
+import '../../core/util/ui_constants.dart';
+import '../views/now_playing_movies_pageview.dart';
+import '../views/popular_movies_grid.dart';
+import '../views/top_rated_movies_list.dart';
+import '../views/upcoming_movies_grid.dart';
+
+class AppDrawer extends StatelessWidget {
+  const AppDrawer({
+    super.key,
+  });
+
+  static const String drawerHeader = 'What do you feel like watching?';
+  static const String popular = 'Popular movies';
+  static const String topRated = 'Top rated movies';
+  static const String upcoming = 'Upcoming movies';
+  static const String nowPlaying = 'Now playing movies';
+
+  @override
+  Widget build(BuildContext context) {
+    return Drawer(
+      backgroundColor: UIConstants.backgroundColor,
+      child: ListView(
+        children: [
+          const DrawerHeader(
+            decoration: BoxDecoration(
+              color: UIConstants.backgroundColor,
+            ),
+            child: Align(
+              alignment: Alignment.centerLeft,
+              child: Text(
+                drawerHeader,
+                style: UIConstants.titleTextStyle,
+              ),
+            ),
+          ),
+          ListTile(
+            title: const Text(
+              popular,
+              style: UIConstants.drawerTextStyle,
+            ),
+            onTap: () {
+              Navigator.pop(context);
+              Navigator.pushNamed(
+                context,
+                PopularMoviesGrid.routeName,
+              );
+            },
+          ),
+          ListTile(
+            title: const Text(
+              topRated,
+              style: UIConstants.drawerTextStyle,
+            ),
+            onTap: () {
+              Navigator.pop(context);
+              Navigator.pushNamed(
+                context,
+                TopRatedMoviesList.routeName,
+              );
+            },
+          ),
+          ListTile(
+            title: const Text(
+              upcoming,
+              style: UIConstants.drawerTextStyle,
+            ),
+            onTap: () {
+              Navigator.pushNamed(
+                context,
+                UpcomingMoviesGrid.routeName,
+              );
+              //Navigator.pop(context);
```

Comment:
remove this if not needed

Reply:
yes, I had to put that line before Navigator.pushNamed!
done